### DISCUSSION TITLE
Predictions against the spread (ATS)

### DISCUSSION THREAD

**Discussion by JimCollinsDC** [kyleskom/NBA-Machine-Learning-Sports-Betting#89] [Ideas]
how hard would it be to have this use ATS numbers in its predictions? So far it only seems to use money line unless I'm missing something.

**Comment by kyleskom** [comment]
Thats correct. Adding is spread wouldn't be to hard at all.

**Reply by nealmick** [reply]
Hey kyle,
Just wanted to thank you again for the data you've collected.  In the past two weeks ive converted my model to predict scores in-stead of a sigmoid function.  Learned that from your project.   I then was able to extract the games won and games lost features from your datasets, as-well as the point spread.  Both features ive been wanting to add for a long time, and it significantly boosted accuracy.  The next features im after are win/loss streak and team away / home data instead of over all.
Ive read here that using xgboost improved accuracy, how important do you think this is?
Also if you had unlimited api data what features do you think are the most important?

Thanks a bunch

**Reply by kyleskom** [reply]
I don't think using xgboost is that important, I just found it to be slightly better when I trained awhile back. If anything else could be added it would be someway to incorporate player data as well.

**Reply by nealmick** [reply]
Thanks for the info!! I do have individual player level data.  Season averages and specific game stats going back a while(pts,reb,stl,min.....).  Currently for each team i am using the top 7 players by playtime.  Also i just had this thought I extracted your data but could probably put something together to do the opposite and put a data-set with player level data on top your data.

Cheers, Neal

**Reply by kyleskom** [reply]
If you could somehow incorporate / add player data to this that would be extremely welcomed.

**Reply by JimCollinsDC** [reply]
nba-api has player data. Discuss on our slack channel https://nbaapi.slack.com/

**Reply by kyleskom** [reply]
Im not worried about getting player data, tricky part is incorporating it for every game. How to combine player data with game data for all games going back over a decade.

**Reply by JimCollinsDC** [reply]
IMHO game data from a decade ago is irrelevant; game has changed. Decade ago is the ancient past.
Is there a way to give more recent seasons more weight?

**Reply by JimCollinsDC** [reply]
does model take into account if they played a game the night before? esp an away game? Travel? very important.

**Reply by kyleskom** [reply]
Home team and away team should be accounted for. We are working on adding rest days currently. 

The point wasn't about going back a decade. The challenge of adding player data to past games to train with isn't trivial.

**Reply by nealmick** [reply]
Built a little thing to merge data.
https://github.com/nealmick/Merge-Data

Probably a few bugs still but seems to be working.

**Reply by kyleskom** [reply]
Does this just get player data or does it add those players to a game as well? So say Chi played LA. Does it add player data to the team data for a given game?

**Reply by nealmick** [reply]
Well tbh its kinda in-between.  It adds the data to the Team-Data folder.  But it will only add data if a game actually occurred on the date.  all the data added is derived from real games.  say chi vs la on 2015-10-1 (not real game).  I gather the data like this, seasons 
 and teams determine the rosters, they are saved in PlayerIDbyTeamId, then using the players I get there season averages, then sort and so on....

**Reply by kyleskom** [reply]
If you don't mind lets revisit this ideas after the few pr's we have coming up.

**Reply by kyleskom** [reply]
Im def interested in this and plater data is the next major thing.

**Reply by nealmick** [reply]
Absolutely,  Let me know,  I kinda owe you some of my data after using all of yours!
